extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -66,18 +59,17 @@ export function Scroller({
 
   // useLayoutEffect needed to avoid FOUC while scrolling
   useLayoutEffect(() => {
-    if (!containerRef.current) {
+    if (!ref.current) {
```

Comment:
So this is actually more complicated. We cannot rely on `ref` to actually be there. The component basically says "I'll forward ref when I'm asked to". But if no one asks - you will get `ref == null`. So, instead, I think you would need to do something like this:

```
function ScrollerWithRef(props, ref) {
  // We still need our own ref that we can always rely on to be there.
  const containerRef = useRef();
  useImperativeHandle(ref, () => containerRef.current);

  useEffect(() => {
   // We use "own our" containerRef here.
    if (containerRef.current) {...}
  });

   // We use "own our" containerRef here.
  return <div ref={containerRef}>...;
}
```

There's a slight variation to this if we could use function-based ref. But I think the way above is better for now.

Reply:
Thanks for clarifying this nuance. I restored all references to `containerRef`. Additionally instead of having `useImperativeHandle` reach in and returning `containerRef.current`, I instead expose `advance` so we can more cleanly do this:
```js
  const scrollRef = useRef(null);
  const advance = (by) => scrollRef.current.advance(by);
```

Reply:
This looks good!